src/urllib3/multipart/encoder.py

```diff
@@ -0,0 +1,676 @@
+"""
+
+requests_toolbelt.multipart.encoder
+===================================
+
+This holds all of the implementation details of the MultipartEncoder
+
+"""
+import contextlib
+import io
+import os
+import typing
+import uuid
+
+from .. import fields
+
+
+class FileNotSupportedError(Exception):
+    """The file provided by the user is not supportable."""
+
+
+P = typing.TypeVar("P", bound="Part")
+
+
+class Part:
```

Comment:
I don't think we can since we _want_ to cache the length of the part and compute it and I don't think it makes sense to have a tuple that has that as an attribute since you're so opposed to the idea of having an attribute for that.